File: www/node/test/tenki.py
# 天気予報を取得するモジュール
# URLからJSONデータをダウンロードする関数 --- (*1)
def get_json(url):
    import urllib.request as req
    import json
    # データをダウンロード
    res = req.urlopen(url)
    json_data = res.read()
    # JSONデータをPythonで扱えるよう読み込む
    return json.loads(json_data)

# 都市IDを取得することで天気情報を取得する関数 --- (*2)
def get_weather(city_id):
    url = "http://weather.livedoor.com/forecast/webservice/json/v1"
    url += "?city=" + str(city_id)
    data = get_json(url)
    s = ""
    for row in data["forecasts"]:
        label = row["dateLabel"]
        telop = row["telop"]
        s += label + " : " + telop + "\n"
    return s

# モジュールとして利用されるかどうか判定
if __name__ == "__main__":
    res = get_weather(130010) # 東京の天気を取得 --- (*3)
    print(res)

File: www/pyjstest/tenki.js
define('__main__',function (require,exports,module) {
  var PYLIB=require('http://localhost/runtime/lib/python/PyLib.js');
  var __top=PYLIB.moduleScope(PYLIB, false);
  PYLIB.LoopChecker.reset();
  
  
  __top.get_json=function (url){
    let req,json,res,json_data;
    req=require('http://localhost/runtime/lib/python/py_urllib,request.js').install(PYLIB);
    json=require('http://localhost/runtime/lib/python/py_json.js').install(PYLIB);
    
    res=req.__getattribute__('urlopen')(url);
    json_data=res.__getattribute__('read')();
    
    return json.__getattribute__('loads')(json_data);
  };
  
  
  __top.get_weather=function (city_id){
    let url,data,s,row,label,telop;
    url="http://weather.livedoor.com/forecast/webservice/json/v1";
    url=(url).__iadd__(("?city=").__add__(__top.str(city_id)))
    data=__top.get_json(url);
    s="";
    for (row of data.__getitem__("forecasts")) {
      PYLIB.LoopChecker.check();
      label=row.__getitem__("dateLabel");
      telop=row.__getitem__("telop");
      s=(s).__iadd__((((label).__add__(" : ")).__add__(telop)).__add__("\n"))
    }return s;
  };
  
  
  if ((__top.__name__).__eq__("__main__")) {
    PYLIB.LoopChecker.check();
    __top.res=__top.get_weather(130010);
    PYLIB.print(__top.res);
  }
});
requirejs(['__main__'],function(){
            if (window.parent && window.parent.sendResult) {
                window.parent.sendResult($("#output").text(),"py");
            }});
